backend/tests/evals/test_e2e.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any
+
+import pandas as pd
+from langchain_anthropic import ChatAnthropic
+from langchain_core.documents import Document
+from langchain_core.messages import AIMessage
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.pydantic_v1 import BaseModel, Field
+from langsmith.evaluation import EvaluationResults, evaluate
+from langsmith.schemas import Example, Run
+
+from backend.graph import OPENAI_MODEL_KEY, format_docs, graph
+
+DATASET_NAME = "chat-langchain-qa"
+EXPERIMENT_PREFIX = "chat-langchain-ci"
+
+SCORE_RETRIEVAL_RECALL = "retrieval_recall"
+SCORE_ANSWER_CORRECTNESS = "answer_correctness_score"
+SCORE_ANSWER_VS_CONTEXT_CORRETNESS = "answer_vs_context_correctness_score"
+
+# claude sonnet / gpt-4o are a bit too expensive
+JUDGE_MODEL_NAME = "claude-3-haiku-20240307"
```

Comment:
ideally of course it's something like 3.5 sonnet, but it's a bit pricey at ~$1 or so per eval run

Reply:
Ya haiku should be fine - can check correlation and self-improve if we want

Reply:
We could also just configure these in-app on the dataset and add some self-improvement